libmogwai-schedule/connection-monitor-nm.c

```diff
@@ -615,16 +615,16 @@ mws_connection_monitor_nm_get_connection_details (MwsConnectionMonitor *monitor,
   /* TODO: If we want to load a default value from eos-autoupdater.conf (see
    * https://phabricator.endlessm.com/T20818#542708), we should plumb it in here
    * (but do the async loading of the file somewhere else, earlier). */
-  const gboolean download_when_metered_default = FALSE;
+  const gboolean allow_downloads_when_metered_default = FALSE;
```

Comment:
Wasn't the default value supposed to be `TRUE`?

The Automatic Updates panel now unconditionally sets it to `'1'` now.

Reply:
Ignoring the whole system and just looking at Mogwai, I think the default for `allow-downloads-when-metered` should be `FALSE` because the alternative is potentially expensive for users.

Looking at the whole system, the code in g-c-c will set its default to be `1` if the connection is unmetered, which is equivalent to:
 * `connection.allow-downloads` = `1`
 * `connection.allow-downloads-when-metered` = `0`
which is exactly what Mogwai does here.

So I don’t think there’s a problem, but please correct me if I’m wrong!

Reply:
I am confused now. From the 24-04 meeting notes, I read:

 * Keep the existing `allow-downloads-when-metered` NM key, and keep it
per-connection.
 • Decouple it from the UI and set it to always be `TRUE` for the moment.
We may recouple it to the UI in future, as we expose more functionality
in the g-c-c UI.

Which is precisely what Control Center is doing now. It changes `connection.allows-downloads`, and unconditionally sets `connection.allows-downloads-when-metered` to `'1'` (i.e. `TRUE`).

Reply:
Well, it might be the case that Mogwai doesn't care about what GNOME Settings do. But I reinforce that, from what I undertood from the meeting notes, the default value of `connection.allows-downloads-when-metered` should be `TRUE`. These rules (TRUE if connection is not metered, FALSE otherwise) now apply to `connection.allows-downloads`'s default value.

Reply:
What g-c-c is doing is correct. The code you’ve commented on only affects the default values of the settings (i.e. if g-c-c has not been used to change the settings yet).

My point is that these rules (“TRUE if connection is not metered, FALSE otherwise”) applying to `connection.allow-downloads` (what g-c-c does) is equivalent to setting `connection.allow-downloads-when-metered` to `0` and `connection.allow-downloads` to `1` (what Mogwai does if neither setting is set).

Reply:
Ok then. This discussion is resolved.